src/tosyntax/FStar.ToSyntax.ToSyntax.fs

```diff
@@ -636,7 +636,17 @@ and desugar_typ env e : S.term =
     desugar_term_maybe_top false env e
 
 and desugar_machine_integer env repr (signedness, width) range =
-  let lower, upper = FStar.Const.bounds signedness width in
+  let size = match width with
+             | Int8  -> 8
+             | Int16  -> 16
+             | Int32  -> 32
+             | Int64  -> 64 in
+  let lower, upper = begin match signedness with
+                           | Unsigned -> 0I, System.Numerics.BigInteger.Pow(2I, size) - 1I
+                           | Signed -> 
+                              let upper' = System.Numerics.BigInteger.Pow(2I, size - 1) in
+                              -upper' , upper' - 1I
```

Comment:
Doesn't this break extraction to OCaml? That file (`FStar.ToSyntax.ToSyntax.fs`) is supposed to be valid F*, IIUC.

Reply:
My mistake, I will remove this. It's probably best to fix the F# build to use bigints throughout, like the OCaml build.

Reply:
I think your help would be welcome on this, but there's a rich history on that issue :) https://github.com/FStarLang/FStar/pull/1085